src/io/file_descriptor.cr

```diff
@@ -131,7 +131,13 @@ class IO::FileDescriptor
     check_open
 
     flush
-    seek_value = LibC.lseek(@fd, offset, whence)
+
+    if whence.current?
+      seek_value = LibC.lseek(@fd, tell + offset, Seek::Set)
```

Comment:
Would it not be acceptable simply to remove `@in_buffer_rem.size` from the `offset` here and not use `Seek::Set`?